Projects/Checkers/Board.cs

```diff
@@ -0,0 +1,171 @@
+				new() { NotationPosition ="B6", Color = White},
+				new() { NotationPosition ="C7", Color = White},
+				new() { NotationPosition ="D8", Color = White},
+				new() { NotationPosition ="D6", Color = White},
+				new() { NotationPosition ="E7", Color = White},
+				new() { NotationPosition ="F8", Color = White},
+				new() { NotationPosition ="F6", Color = White},
+				new() { NotationPosition ="G7", Color = White},
+				new() { NotationPosition ="H8", Color = White},
+				new() { NotationPosition ="H6", Color = White}
+			};
+	}
+
+	public static string ToPositionNotationString(int x, int y)
+	{
+		if (!IsValidPosition(x, y)) throw new ArgumentException("Not a valid position!");
+		return $"{(char)('A' + x)}{y + 1}";
+	}
+
+	public static (int X, int Y) ParsePositionNotation(string notation)
+	{
+		if (notation is null) throw new ArgumentNullException(nameof(notation));
+		notation = notation.Trim().ToUpper();
+		if (notation.Length is not 2 ||
+			notation[0] < 'A' || 'H' < notation[0] ||
+			notation[1] < '1' || '8' < notation[1])
+			throw new FormatException($@"{nameof(notation)} ""{notation}"" is not valid");
+		return (notation[0] - 'A', notation[1] - '1');
+	}
+
+	public static bool IsValidPosition(int x, int y) =>
+		0 <= x && x < 8 &&
+		0 <= y && y < 8;
+
+	public (Piece A, Piece B) GetClosestRivalPieces(PieceColor priorityColor)
+	{
+		double minDistanceSquared = double.MaxValue;
+		(Piece A, Piece B) closestRivals = (null!, null!);
+		foreach (Piece a in Pieces.Where(piece => piece.Color == priorityColor))
+		{
+			foreach (Piece b in Pieces.Where(piece => piece.Color != priorityColor))
+			{
+				(int X, int Y) vector = (a.X - b.X, a.Y - b.Y);
+				double distanceSquared = vector.X * vector.X + vector.Y * vector.Y;
+				if (distanceSquared < minDistanceSquared)
+				{
+					minDistanceSquared = distanceSquared;
+					closestRivals = (a, b);
+				}
+			}
+		}
+		return closestRivals;
+	}
+
+	public List<Move> GetPossibleMoves(PieceColor color)
+	{
+		List<Move> moves = new();
+		if (Aggressor is not null)
+		{
+			if (Aggressor.Color != color)
+			{
+				throw new Exception($"{nameof(Aggressor)} is not null && {nameof(Aggressor)}.{nameof(Aggressor.Color)} != {nameof(color)}");
+			}
+			moves.AddRange(GetPossibleMoves(Aggressor).Where(move => move.PieceToCapture is not null));
+		}
+		else
+		{
+			foreach (Piece piece in Pieces.Where(piece => piece.Color == color))
+			{
+				moves.AddRange(GetPossibleMoves(piece));
+			}
+		}
+		return moves.Any(move => move.PieceToCapture is not null)
+			? moves.Where(move => move.PieceToCapture is not null).ToList()
+			: moves;
+	}
+
+	public List<Move> GetPossibleMoves(Piece piece)
+	{
+		List<Move> moves = new();
+		ValidateMove(-1, -1);
+		ValidateMove(-1, 1);
+		ValidateMove(1, -1);
+		ValidateMove(1, 1);
+		return moves.Any(move => move.PieceToCapture is not null)
+			? moves.Where(move => move.PieceToCapture is not null).ToList()
+			: moves;
+
+		void ValidateMove(int dx, int dy)
```

Comment:
There's another ValidateMove method later on in the module. Presumably one is for player moves and the other for the computer. Might be worth naming them thusly.

Reply:
hopefully a7f2c78fffdd90aa5ffd754afc7577903dcf9b37 fixes this in your opinion